src/main/fc/fc_core.c

```diff
@@ -485,7 +483,9 @@ void processRx(timeUs_t currentTimeUs)
             DISABLE_FLIGHT_MODE(HEADFREE_MODE);
         }
         if (IS_RC_MODE_ACTIVE(BOXHEADADJ)) {
-            headFreeModeHold = DECIDEGREES_TO_DEGREES(attitude.values.yaw); // acquire new heading
```

Comment:
No it is not identical ...
HEADFREE_MODE is active 
and not disabled 
HEADFREE_MODE code  code gets executed !

You have a strange interpretation.